src/Stratis.Bitcoin.Features.ColdStaking/ColdStakingFeature.cs

```diff
@@ -183,19 +181,40 @@ private void AddComponentStats(StringBuilder benchLog)
 
                 foreach (string walletName in walletNames)
                 {
-                    IEnumerable<UnspentOutputReference> normalTransactions = this.coldStakingManager.GetSpendableTransactionsInWallet(walletName, 1);
-                    IEnumerable<UnspentOutputReference> hotWalletTransactions = this.coldStakingManager.GetSpendableTransactionsInColdWallet(walletName, false, 1);
-                    IEnumerable<UnspentOutputReference> coldWalletTransactons = this.coldStakingManager.GetSpendableTransactionsInColdWallet(walletName, true, 1);
-                    benchLog.AppendLine("Wallet     : " + (walletName + ",").PadRight(LoggingConfiguration.ColumnLength) + " Confirmed balance: " + new Money(normalTransactions.Sum(s => s.Transaction.Amount)).ToString() + " Cold stake hot balance: " + new Money(hotWalletTransactions.Sum(s => s.Transaction.Amount)).ToString() + " Cold stake cold balance: " + new Money(coldWalletTransactons.Sum(s => s.Transaction.Amount)).ToString());
+                    // Get all the accounts, including the ones used for cold staking.
+                    // TODO: change GetAccounts to accept a filter.
+                    foreach (HdAccount account in this.coldStakingManager.GetAccounts(walletName))
```

Comment:
Yes, perhaps GetAccounts can be changed in which case less code would be needed:

```
        public IEnumerable<HdAccount> GetAccounts(string walletName, Func<HdAccount, bool> accountFilter)
        {
            Guard.NotEmpty(walletName, nameof(walletName));

            Wallet wallet = this.GetWalletByName(walletName);

            HdAccount[] res = null;
            lock (this.lockObject)
            {
                res = wallet.GetAccountsByCoinType(this.coinType, accountFilter).ToArray();
            }
            return res;
        }
```

then the call becomes:

`foreach (HdAccount account in this.coldStakingManager.GetAccounts(walletName, a => true))`

to return all accounts.

Reply:
Indeed. I have a bunch of changes similar to that in another branch.
Coming soon, maybe not before release.